Classes/Controller/GetFileController.php

```diff
@@ -95,48 +99,74 @@ public function attachmentAction()
 
                     header('Content-Type: text/xml; charset=UTF-8');
 
-                }
+                    return $metsXml;
+
+                } else {
+
+                    http_response_code(404);
 
-                return $metsXml;
+                    print('No such document');
+
+                    exit;
```

Comment:
Is `exit` really this what you want? I think a `return` statement is a better solution.